### ISSUE TITLE
About Gated Query Transformation

### ISSUE THREAD

**Issue by 123456789asdfjkl** [WillDreamer/Aurora#2]
Hi! Thanks for your great work! I want to know how the gated mechanism is implemented？
![image](https://github.com/WillDreamer/Aurora/assets/45223476/cdfb3d52-6d3b-4f3b-8d60-5da61bc00fa1)

**Comment by xinlong-yang** [comment]
Thanks for your interest. After transforming t to t' through a scale&shift operation, we get paired (f, t') which has shape (seq_len, hidden_dim), then **g** can be computed through codes like following:
`g = torch.softmax(torch.sum(torch.matmul(f,t'.T),dim=1),dim=0).unsqueeze_(1)`

**Comment by 123456789asdfjkl** [comment]
I see. Thanks for your answer！

**Comment by Eureka-JTX** [comment]
@123456789asdfjkl Hi，have you reproduced it successfully？

**Comment by Arsiuuu** [comment]
@123456789asdfjkl Hello~ have you reproduced successfully?
